projects/aca-content/src/lib/dialogs/folder-details/folder-information.component.ts

```diff
@@ -0,0 +1,110 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component, DestroyRef, inject, OnInit, ViewEncapsulation } from '@angular/core';
+import { CommonModule, NgIf } from '@angular/common';
+import { DIALOG_COMPONENT_DATA, LocalizedDatePipe, TimeAgoPipe } from '@alfresco/adf-core';
+import { JobIdBodyEntry, Node, SizeDetailsEntry } from '@alfresco/js-api';
+import { MatDividerModule } from '@angular/material/divider';
+import { TranslateModule, TranslateService } from '@ngx-translate/core';
+import { ContentService, NodesApiService } from '@alfresco/adf-content-services';
+import { concatMap, expand, first } from 'rxjs/operators';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { EMPTY, timer } from 'rxjs';
+
+const MEMORY_UNIT_LIST = ['bytes', 'KB', 'MB', 'GB', 'TB'];
+
+interface FolderDetails {
+  name: string;
+  size: string;
+  location: string;
+  created: Date;
+  modified: Date;
+  icon: string;
+}
+
+@Component({
+  selector: 'app-folder-info',
+  standalone: true,
+  imports: [CommonModule, MatDividerModule, TimeAgoPipe, NgIf, TranslateModule, LocalizedDatePipe, LocalizedDatePipe],
+  templateUrl: './folder-information.component.html',
+  styleUrls: ['./folder-information.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  host: { class: 'app-folder-info' }
+})
+export class FolderInformationComponent implements OnInit {
+  readonly contentService = inject(ContentService);
+  readonly nodesService = inject(NodesApiService);
+  readonly translateService = inject(TranslateService);
+
+  private readonly destroyRef = inject(DestroyRef);
+
+  data: Node = inject(DIALOG_COMPONENT_DATA);
+  folderDetails: FolderDetails;
+
+  ngOnInit() {
+    this.folderDetails.name = this.data.name;
+    this.folderDetails.location = this.data.path.name;
+    this.folderDetails.created = this.data.createdAt;
+    this.folderDetails.modified = this.data.modifiedAt;
+    this.folderDetails.icon = this.contentService.getNodeIcon(this.data);
+    this.folderDetails.size = this.translateService.instant('APP.FOLDER_INFO.CALCULATING');
+
+    this.nodesService
+      .initiateFolderSizeCalculation(this.data.id)
+      .pipe(first())
+      .subscribe((jobIdEntry: JobIdBodyEntry) => {
+        this.nodesService
+          .getFolderSizeInfo(this.data.id, jobIdEntry.entry.jobId)
+          .pipe(
+            expand((result: SizeDetailsEntry) =>
+              result.entry.status === 'IN_PROGRESS'
+                ? timer(5000).pipe(concatMap(() => this.nodesService.getFolderSizeInfo(this.data.id, jobIdEntry.entry.jobId)))
+                : EMPTY
+            ),
+            takeUntilDestroyed(this.destroyRef)
+          )
+          .subscribe((folderInfo: SizeDetailsEntry) => {
```

Comment:
is that possible to use some operator to avoid subscribe inside other subscribe?

Reply:
I've updated the code to use switchMap. However, I still need to test it in some scenarios. Will update here if any changes are done